frontend/src/AuthPage.js

```diff
@@ -1,57 +1,58 @@
-import Cookies from 'js-cookie';
-
-import NavBar from './NavBar';
-import Footer from './Footer';
-import ErrorBox from './ErrorBox';
-import ReviewPage from './ReviewPage';
-import { redirect_for_auth, api_is_authenticated } from './api';
+import React, { Component } from 'react'
+import Cookies from 'js-cookie'
 
+import NavBar from './NavBar'
+import Footer from './Footer'
+import ErrorBox from './ErrorBox'
+import ReviewPage from './ReviewPage'
+import { redirect_for_auth, api_is_authenticated } from './api'
 
 /**
  * A wrapper around a review page that performs Shibboleth authentication.
  */
 class AuthPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isAuthed: false,
-            authFailed: false
-        };
+  constructor(props) {
+    super(props)
 
-        const temp_cookie_name = 'doing_token_auth';
-        api_is_authenticated((authed) => {
-            if (authed) {
-                Cookies.remove(temp_cookie_name);
-                this.setState({ isAuthed: true, authFailed: false });
-            }
-            else {
-                if (typeof Cookies.get(temp_cookie_name) === 'undefined') {
-                    Cookies.set(temp_cookie_name, 'true', { expires: 1/1440 });
-                    redirect_for_auth();
-                    this.setState({ isAuthed: false, authFailed: false });
-                }
-                else {
-                    Cookies.remove(temp_cookie_name);
-                    this.setState({ isAuthed: false, authFailed: true });
-                }
-            }
-        });
+    this.state = {
+      isAuthed: false,
+      authFailed: false,
     }
 
-    render() {
-        if (this.state.authFailed) {
-            return <div>
-                <NavBar />
-                <ErrorBox>Could not perform Shibboleth authentication.<br />Refresh this page to try again.</ErrorBox>
-                <Footer />
-            </div>;
-        }
-
-        return this.state.isAuthed ? <ReviewPage {...this.props} /> : null;
+    const tempCookieName = 'doing_token_auth'
+    api_is_authenticated((authed) => {
+      if (authed) {
+        Cookies.remove(tempCookieName)
+        this.setState({ isAuthed: true, authFailed: false })
+      } else if (typeof Cookies.get(tempCookieName) === 'undefined') {
+        Cookies.set(tempCookieName, 'true', { expires: 1 / 1440 })
+        redirect_for_auth()
+        this.setState({ isAuthed: false, authFailed: false })
+      } else {
+        Cookies.remove(tempCookieName)
+        this.setState({ isAuthed: false, authFailed: true })
+      }
+    })
+  }
+
+  render() {
+    const { authFailed, isAuthed } = this.state
+    if (authFailed) {
+      return (
+        <div>
+          <NavBar />
+          <ErrorBox>
+            Could not perform Shibboleth authentication.
+            <br />
+            Refresh this page to try again.
+          </ErrorBox>
+          <Footer />
+        </div>
```

Comment:
Spin this out to its own component?

Reply:
It's not being used anywhere else so I'd rather not for now, components and pages will be refactored during the upcoming port to nextjs